website/src/pages/documents/document.page.tsx

```diff
@@ -493,23 +489,101 @@ export const DocumentTitleHeader = (p: {
-          <span>Document Audio:</span>
-          <AudioPlayer
-            style={{ flex: 1 }}
-            audioUrl={p.doc.audioRecording.resourceUrl}
-            showProgress
-          />
-          {p.doc.audioRecording && !isMobile && (
-            <div>
-              <a href={p.doc.audioRecording?.resourceUrl}>
-                <Button>Download Audio</Button>
-              </a>
-            </div>
+      <div id="audio-and-recording-container">
+        <div>
+          {/* only show document audio to non-editors , editors can see which ones are shown from checkmark*/}
+          {p.doc.editedAudio.length > 0 && role !== UserRole.Editor && (
+            <>
+              <h3>Document Audio:</h3>
+              {p.doc.editedAudio.map((audio, index) => (
+                <div
+                  id={`document-audio-player-${index}`}
+                  className={css.audioContainer}
+                  key={index}
+                >
+                  <AudioPlayer
+                    contributor={audio.recordedBy?.displayName}
+                    recordedAt={
+                      audio.recordedAt?.formattedDate
+                        ? new Date(audio.recordedAt.formattedDate)
+                        : undefined
+                    }
+                    style={{ flex: 1 }}
+                    audioUrl={audio.resourceUrl}
+                    showProgress
+                  />
+                  {!isMobile && (
+                    <div>
+                      <a href={audio.resourceUrl}>
+                        <Button>Download Audio</Button>
+                      </a>
+                    </div>
+                  )}
+                </div>
+              ))}
+            </>
+          )}
+          {role === UserRole.Contributor && (
+            <>
+              <h3>User-contributed Audio:</h3>
+              {p.doc.userContributedAudio.map(
+                (audio, index) =>
+                  audio.recordedBy?.id === userId && (
+                    <div key={index}>
+                      <AudioPlayer
+                        contributor={"you"}
+                        recordedAt={
+                          audio.recordedAt?.formattedDate
+                            ? new Date(audio.recordedAt.formattedDate)
+                            : undefined
+                        }
+                        audioUrl={audio.resourceUrl}
+                        showProgress
+                      />
+                    </div>
+                  )
+              )}
+            </>
+          )}
+          {role === UserRole.Editor && (
+            <>
+              <h3>User-contributed Audio:</h3>
+              {p.doc.userContributedAudio.map((audio, index) => (
+                <div
+                  id={`user-contributed-document-audio-player-${index}`}
+                  className={css.documentAudioContainer}
+                  key={index}
+                >
+                  {role === UserRole.Editor && (
```

Comment:
Since you're already inside a UserRole.Editor block I think this second check is redundant.

Reply:
agreed– is there a reason you wrote it this way Dennis?